contributing.md

```diff
@@ -0,0 +1,277 @@
+# Contribution workflow and the review process
+
+This document outlines the best practice, using git and CI, which
+**must** be followed for all contributions to janus-core. Also contained
+are instructions and tips for managing your fork of the project which
+will help keep merges clean and avoid many headaches.
+
+## Golden rules
+
+In brief the rules for contribution are as follows:
+
+-   Follow the branch, fix, merge model, from your own fork or fork/branch model.
+-   An issue must be created for every piece of work (bug, feature,
+    *etc.*)
+-   Pull/merge requests will not be accepted without a review
+-   New features must have a test
+-   All tests must pass, no regressions may be merged
+
+## Issues
+
+### Using issues
+
+-   Open an issue for each piece of work done.
+-   Open issues for design discussions. The answers may be important for
+    newer members.
+-   New features, *e.g.* add new MLIP, comments should be used to
+    provide succint reports on progress.
```

Comment:
```suggestion
-   When adding features, *e.g.* a new MLIP, use comments to provide succinct reports on progress.
```

Reply:
Yep, just didn't really seem long enough to split it into two in this case when I was rewording it, which was the main point, happy for it to be two still.

Also yes, I previously was getting rid of the extra spaces, but I'd rather be consistent and didn't want to change all of them...